lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_lint_localizations.rb

```diff
@@ -49,18 +48,21 @@ def self.report(violations:, base_lang:)
       def self.find_duplicated_keys(params)
         duplicate_keys = {}
 
-        files_to_lint = Dir.chdir(params[:input_dir]) do
-          Dir.glob('*.lproj/Localizable.strings').map do |file|
-            {
-              language: File.basename(File.dirname(file), '.lproj'),
-              path: File.join(params[:input_dir], file)
-            }
-          end
-        end
-
+        files_to_lint = Dir.glob('*.lproj/Localizable.strings', base: params[:input_dir])
         files_to_lint.each do |file|
-          duplicates = Fastlane::Helper::Ios::StringsFileValidationHelper.find_duplicated_keys(file: file[:path])
-          duplicate_keys[file[:language]] = duplicates.map { |key, value| "`#{key}` was found at multiple lines: #{value.join(', ')}" } unless duplicates.empty?
+          language = File.basename(File.dirname(file), '.lproj')
+          path = File.join(params[:input_dir], file)
+
+          file_type = Fastlane::Helper::Ios::L10nHelper.strings_file_type(path: path)
+          if file_type == :text
+            duplicates = Fastlane::Helper::Ios::StringsFileValidationHelper.find_duplicated_keys(file: path)
+            duplicate_keys[language] = duplicates.map { |key, value| "`#{key}` was found at multiple lines: #{value.join(', ')}" } unless duplicates.empty?
+          else
+            UI.warning <<~WRONG_FORMAT
+              File `#{path}` is in #{file_type} format, while finding duplicate keys only make sense on files that are in ASCII-plist format.
+              Since your files are in #{file_type} format, you should probably disable the `check_duplicate_keys` option from this `#{self.action_name}` call.
+            WRONG_FORMAT
```

Comment:
As suggested by @leandroalonso in https://github.com/wordpress-mobile/release-toolkit/issues/405#issuecomment-1267270031